tests/datasourceextensions.test.ts

```diff
@@ -250,13 +231,13 @@ export class DBOSKnexDS implements DBOSTransactionalDataSource {
                 //  1. The transaction is marked failed, but the user code did not throw.
                 //      Bad on them.  We will throw an error (this will get recorded) and not retry.
                 //  2. There was a key conflict in the statement, and we need to use the fetched output
-                if (this.isFailedSqlTransactionError(error)) {
+                if (isPGFailedSqlTransactionError(error)) {
                   DBOS.logger.error(
                     `In workflow ${wfid}, Postgres aborted a transaction but the function '${funcname}' did not raise an exception.  Please ensure that the transaction method raises an exception if the database transaction is aborted.`,
                   );
                   failedForRetriableReasons = false;
                   throw new DBOSFailedSqlTransactionError(wfid, funcname);
-                } else if (this.isKeyConflictError(error)) {
+                } else if (isPGKeyConflictError(error)) {
```

Comment:
Note, we've decided we do need to record errors (regardless of the comment a little bit further down says). We should update this test at somepoint to follow current plan